packages/wonder-blocks-form/src/components/__tests__/labeled-text-field.test.js

```diff
@@ -37,11 +37,103 @@ describe("LabeledTextField", () => {
         expect(wrapper).toHaveState("focused", false);
     });
 
-    it("disabled prop disables the input", async () => {
+    it("value state changes when the user types", () => {
+        // Arrange
+        const wrapper = mount(<LabeledTextField label="Label" />);
+        const input = wrapper.find("input");
+
+        // Act
+        const newValue = "New Value";
+        input.simulate("change", {target: {value: newValue}});
+
+        // Assert
+        expect(wrapper).toHaveState("value", newValue);
+    });
+
+    it("id prop is passed to input", () => {
+        // Arrange
+        const id = "exampleid";
+
+        // Act
+        const wrapper = mount(
+            <LabeledTextField id={id} label="Label" disabled={true} />,
+        );
+
+        // Assert
+        const input = wrapper.find("input");
+        expect(input).toContainMatchingElement(`[id="${id}-field"]`);
+    });
+
+    it("auto-generated id is passed to input when id prop is not set", () => {
+        // Arrange
+
+        // Act
+        const wrapper = mount(<LabeledTextField label="Label" />);
+
+        // Assert
+        // Since the generated id is unique, we cannot know what it will be
+        // so we only test if the id attribute is set.
+        const input = wrapper.find("input");
+        expect(input).toContainMatchingElement("[id]");
```

Comment:
I think you can also try matching a given pattern. Something like this could probably work.

```suggestion
        expect(input).toMatch(/field-field/);
```

Reply:
I can't seem to get this to work. Does `toMatch(...)` match against the attributes in the HTML element, or against something else?

Reply:
Never mind, I got it working doing something like:
```js
expect(input.props()["id"]).toMatch(/uid-.*-field/);
```